test/sequential/test-watch-mode-inspect.mjs

```diff
@@ -12,26 +12,34 @@ if (common.isIBMi)
 
 common.skipIfInspectorDisabled();
 
+let gettingDebuggedPid = false;
+
+async function getDebuggedPid(instance, waitForLog = true) {
+  gettingDebuggedPid = true;
+  const session = await instance.connectInspectorSession();
+  await session.send({ method: 'Runtime.enable' });
+  if (waitForLog) {
+    await session.waitForConsoleOutput('log', 'safe to debug now');
+  }
+  const { value: innerPid } = (await session.send({
+    'method': 'Runtime.evaluate', 'params': { 'expression': 'process.pid' }
+  })).result;
+  session.disconnect();
+  gettingDebuggedPid = false;
+  return innerPid;
+}
+
 function restart(file) {
   writeFileSync(file, readFileSync(file));
-  const interval = setInterval(() => writeFileSync(file, readFileSync(file)), 500);
+  const interval = setInterval(() => {
+    if (!gettingDebuggedPid) {
```

Comment:
how is this different than
https://github.com/nodejs/node/blob/b8a699ea92761eb0a4c60ba815de07c35dfc729f/test/sequential/test-watch-mode-inspect.mjs#L56

Reply:
It can happen that the line before stopping restarts `pids.push(await getDebuggedPid(instance));` can take some time to finish. Since restarting interval is 500ms it could happen that the restart is triggered while still getting that PID (this can happen even with higher intervals).